### DISCUSSION TITLE
How to upload a file from the client or from postman to the server?

### DISCUSSION THREAD

**Discussion by willantunes** [denoland/deno#24826] [Q&A]
What is the best way to be able to send a file to the database? I see most of people are storing tempFiles in the server and then sending it to the DB. I am trying to achieve the first piece which is the temporary storage, but I am getting errors. This is my server.ts:
`// server.ts
import { Application, Router } from "https://deno.land/x/oak/mod.ts";
import { ensureDir } from "https://deno.land/std/fs/mod.ts";
import { join } from "https://deno.land/std/path/mod.ts";
import { crypto } from "https://deno.land/std/crypto/mod.ts";

const app = new Application();
const router = new Router();

const uploadDir = './uploads';
await ensureDir(uploadDir);

router.post('/upload', async (context) => {
  if (context.request.hasBody) {
    const body = await context.request.body({ type: 'form-data' }).value;
    const rspBody = { fields: 0, files: 0 };

    for (const [name, value] of body.entries()) {
      if (value instanceof File) {
        const newFileName = crypto.randomUUID();
        const filePath = join(uploadDir, newFileName + '-' + value.name);
        await Deno.writeFile(filePath, new Uint8Array(await value.arrayBuffer()));
        rspBody.files++;
      } else {
        rspBody.fields++;
      }
    }

    context.response.body = rspBody;
  } else {
    context.response.status = 400;
    context.response.body = { error: "No file uploaded" };
  }
});

app.use(router.routes());
app.use(router.allowedMethods());

console.log("Server is running on port 8000");
await app.listen({ port: 8000 });`

But I am getting the error i the line 15, it says in the .request.body( ->
This expression is not callable.
  Type 'Body' has no call signatures.deno-ts(2349)
(property) Request.body: Body
An interface to access the body of the request. This provides an API that aligned to the Fetch Request API, but in a dedicated API.

No quick fixes available

Screenshot attached
![image](https://github.com/user-attachments/assets/84b6a5b1-85dc-48eb-b4d6-99f2ff910370)

**Comment by marvinhagemeister** [comment]
According to the [oak documentation](https://jsr.io/@oak/oak/doc/~/Context#property_request) the `context.request` property is not a function, but an instance of the `Request` class. That's why you're getting the error. For Oak specific questions and how to use it, it's best to reach out to the oak project. They have a dedicated GitHub repository where you'll get answers from the folks who wrote oak directly: https://github.com/oakserver/oak/issues?q=sort%3Aupdated-desc+is%3Aissue+is%3Aopen

**Reply by willantunes** [reply]
But does Deno has something to do that in a better way? In a more native way I mean. This was also my question :)
Thanks so far btw

**Reply by marvinhagemeister** [reply]
Oak is pretty close to the native APIs in Deno. Deno itself doesn't come with a router, so going with oak is a solid choice!
